xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-picker/xwiki-platform-attachment-picker-test/xwiki-platform-attachment-picker-test-docker/src/test/it/org/xwiki/attachment/picker/test/AttachmentPickerMacroIT.java

```diff
@@ -0,0 +1,118 @@
+            + "\n"
+            + "{{attachmentPicker id='testPicker3' filter='image/*' /}}\n");
+        AttachmentsPane attachmentsPane = page.openAttachmentsDocExtraPane();
+        attachmentsPane.setFileToUpload(
+            new File(testConfiguration.getBrowser().getTestResourcesPath(), "image1.png").getAbsolutePath());
+        attachmentsPane.waitForUploadToFinish("image1.png");
+        attachmentsPane.setFileToUpload(
+            new File(testConfiguration.getBrowser().getTestResourcesPath(), "image2.png").getAbsolutePath());
+        attachmentsPane.waitForUploadToFinish("image2.png");
+        attachmentsPane.setFileToUpload(
+            new File(testConfiguration.getBrowser().getTestResourcesPath(), "textcontent.txt").getAbsolutePath());
+        attachmentsPane.waitForUploadToFinish("textcontent.txt");
+
+        // Waits for the uploaded files to be indexed before continuing.
+        new SolrTestUtils(setup, computedHostURL(testConfiguration)).waitEmpyQueue();
+
+        // Reload the page to see the file after the uploads and solr indexing.
+        setup.getDriver().navigate().refresh();
+
+        // Test initial display
+        AttachmentPicker testPicker1 = new AttachmentPicker("testPicker1").waitUntilReady();
+        List<String> picker1Attachments = testPicker1.getAttachmentTitles();
+        assertTrue(picker1Attachments.size() >= 3);
+        assertEquals(List.of("image1.png", "image2.png", "textcontent.txt"), picker1Attachments.subList(0, 3));
+
+        // Test on a first filter matching two attachments.
+        testPicker1.setSearch("image").waitUntilAttachmentsCount(2);
+        assertEquals(List.of("image1.png", "image2.png"), testPicker1.getAttachmentTitles());
+
+        // Test on a filter matching no attachment, a warning message is expected.
+        testPicker1.setSearch("doesnotexists").waitUntilAttachmentsCount(0);
+        assertTrue(testPicker1.isNoResultMessageDisplayed());
+
+        // Test with another filter.
+        testPicker1.setSearch("textcontent").waitUntilAttachmentsCount(1);
+        assertEquals(List.of("textcontent.txt"), testPicker1.getAttachmentTitles());
+
+        // Validate that the limit is taken into account.
+        AttachmentPicker testPicker2 = new AttachmentPicker("testPicker2").waitUntilReady();
+        assertEquals(List.of("image1.png", "image2.png"), testPicker2.getAttachmentTitles());
+
+        // Validate that only images are returned.
+        AttachmentPicker testPicker3 = new AttachmentPicker("testPicker3").waitUntilReady();
+        List<String> picker3Attachments = testPicker3.getAttachmentTitles();
+        assertTrue(picker3Attachments.size() >= 2);
+        assertEquals(List.of("image1.png", "image2.png"), picker3Attachments.subList(0, 2));
+        testPicker3.setSearch("textcontent").waitUntilAttachmentsCount(0);
+        assertTrue(testPicker3.isNoResultMessageDisplayed());
+    }
+
+    private String computedHostURL(TestConfiguration testConfiguration)
```

Comment:
It is probably possible to do better for this, but not sure how to get the base url to the wiki from the host.